app/models/server/models/LivechatSessions.js

```diff
@@ -0,0 +1,31 @@
+import { Base } from './_Base';
+
+/**
+ * Livechat session model
+ */
+
+export class LivechatSessions extends Base {
+	constructor() {
+		super('livechat_sessions');
+	}
+
+	getUserLocationByToken(token) {
+		const query = {
+			token,
+		};
+
+		return this.findOne(query);
+	}
+
+	saveLocationForUser(locationData) {
+		const { token, location } = locationData;
+
+		return this.insert({
```

Comment:
```suggestion
		return this.upsert({
```

Reply:
This won't work, throwing an error. Insert is working fine.